pkg/endpoint/endpoint.go

```diff
@@ -933,8 +935,8 @@ func (e *Endpoint) removeDirectory() {
 
 func (e *Endpoint) RemoveDirectory() {
 	e.Mutex.Lock()
+	defer e.Mutex.Unlock()
```

Comment:
`defer` adds a small overhead, leave it as before. For small functions doesn't make sense to use `defer` (IMHO)